submissions/tests/test_team_api.py

```diff
@@ -194,24 +194,58 @@ def test_create_submission_for_team_invalid_attempt(self):
     def test_create_submission_for_team_existing_individual_submission(self):
         """
         Test for calling create_submission_for_team when a user somehow already has
-        an existing active submission for the item.
-
-        For normal Submissions, if a submission exists and we create another one, the second
-        will increment the first's attempt_number. However, for team submissions, we currently
-        pass the attempt_number from team_api.create_submission to api.create_submission, so
-        all created individual submissions will have the same attempt_number
+        an existing active submission for the item in the same course.
+        This can happen is a user was on a team that submitted and then reassigned to a team
+        that didn't have a submission and is submitting.
+        Expected outcome: submission is created because the the previous submission was not team based
         """
         user_3_item = self._get_or_create_student_item(self.anonymous_user_id_map[self.user_3])
         SubmissionFactory.create(student_item=user_3_item)
         team_submission_data = self._call_create_submission_for_team_with_default_args()
         self.assertEqual(team_submission_data['attempt_number'], 1)
-        submissions = Submission.objects.select_related(
+        submission_student_ids = [sub.student_item.student_id for sub in Submission.objects.select_related(
             'student_item'
-        ).filter(
-            uuid__in=team_submission_data['submission_uuids']
-        ).all()
-        for submission in submissions:
-            self.assertEqual(submission.attempt_number, 1)
+        ).filter(uuid__in=team_submission_data['submission_uuids']).all()]
+        self.assertIn(user_3_item.student_id, submission_student_ids)
+
+    def test_create_submission_for_team_existing_individual_submission_different_course(self):
+        """
+        Test for calling create_submission_for_team when a user somehow already has
+        an existing active submission for the item.
+        This can happen is a user was on a team that submitted and then reassigned to a team
+        that didn't have a submission and is submitting.
+        Expected outcome: submission for the user is created because it's a different course
+        """
+        user_3_item = self._get_or_create_student_item(self.anonymous_user_id_map[self.user_3], OTHER_COURSE_ID)
+        SubmissionFactory.create(student_item=user_3_item)
+        team_submission_data = self._call_create_submission_for_team_with_default_args()
+        self.assertEqual(team_submission_data['attempt_number'], 1)
+
+        submission_student_ids = [sub.student_item.student_id for sub in Submission.objects.select_related(
+            'student_item'
+        ).filter(uuid__in=team_submission_data['submission_uuids']).all()]
+        self.assertIn(user_3_item.student_id, submission_student_ids)
+
+    def test_create_submission_for_team_after_reassignment(self):
```

Comment:
This is the real logic we want to test but it's unclear based on how it's written. Why do we expect 4 submission items instead of another number? Since the test functions don't explicitly tell you what the teams/submissions look like, the comments need to be more clear.